aws/resource_aws_sagemaker_model_test.go

```diff
@@ -0,0 +1,709 @@
+resource "aws_sagemaker_model" "foo" {
+	name = "terraform-testacc-sagemaker-model-%s"
+	execution_role_arn = "${aws_iam_role.foo.arn}"
+
+	primary_container {
+		image = "%s"
+		model_data_url = "%s"
+	}
+}
+
+resource "aws_iam_role" "foo" {
+  name = "terraform-testacc-sagemaker-model-%s"
+  path = "/"
+  assume_role_policy = "${data.aws_iam_policy_document.assume_role.json}"
+}
+
+data "aws_iam_policy_document" "assume_role" {
+  statement {
+    actions = [ "sts:AssumeRole" ]
+    principals {
+      type = "Service"
+      identifiers = [ "sagemaker.amazonaws.com" ]
+    }
+  }
+}
+
+resource "aws_iam_policy" "foo" {
+  name = "terraform-testacc-sagemaker-model-%s"
+  description = "Allow Sagemaker to create model"
+  policy = "${data.aws_iam_policy_document.foo.json}"
+}
+
+data "aws_iam_policy_document" "foo" {
+  statement {
+    effect = "Allow"
+    actions = [
+      "cloudwatch:PutMetricData",
+      "logs:CreateLogStream",
+      "logs:PutLogEvents",
+      "logs:CreateLogGroup",
+      "logs:DescribeLogStreams",
+      "ecr:GetAuthorizationToken",
+      "ecr:BatchCheckLayerAvailability",
+      "ecr:GetDownloadUrlForLayer",
+      "ecr:BatchGetImage"
+    ]
+    resources = [
+      "*"]
+  }
+  statement {
+    effect = "Allow"
+    actions = [
+      "s3:GetObject"
+    ]
+    resources = [
+      "arn:aws:s3:::${aws_s3_bucket.foo.bucket}",
+      "arn:aws:s3:::${aws_s3_bucket.foo.bucket}/*"
+    ]
+  }
+}
+
+resource "aws_iam_role_policy_attachment" "foo" {
+  role = "${aws_iam_role.foo.name}"
+  policy_arn = "${aws_iam_policy.foo.arn}"
+}
+
+resource "aws_s3_bucket" "foo" {
+  bucket = "terraform-testacc-sagemaker-model-data-bucket-%s"
+  acl    = "private"
+  force_destroy = true
+}
+
+resource "aws_s3_bucket_object" "object" {
+  bucket = "${aws_s3_bucket.foo.bucket}"
+  key    = "model.tar.gz"
+  content = "some-data"
+}
+`, rName, image, modelDataUrl, rName, rName, rName)
+}
+
+func testAccSagemakerPrimaryContainerHostnameConfig(rName string, image string) string {
+	return fmt.Sprintf(`
+resource "aws_sagemaker_model" "foo" {
+	name = "terraform-testacc-sagemaker-model-%s"
+	execution_role_arn = "${aws_iam_role.foo.arn}"
+
+	primary_container {
+		image = "%s"
+		container_hostname = "foo"
+	}
+}
+
+resource "aws_iam_role" "foo" {
+  name = "terraform-testacc-sagemaker-model-%s"
+  path = "/"
+  assume_role_policy = "${data.aws_iam_policy_document.assume_role.json}"
+}
+
+data "aws_iam_policy_document" "assume_role" {
+  statement {
+    actions = [ "sts:AssumeRole" ]
+    principals {
+      type = "Service"
+      identifiers = [ "sagemaker.amazonaws.com" ]
+    }
+  }
+}
+`, rName, image, rName)
+}
+
+func testAccSagemakerPrimaryContainerEnvironmentConfig(rName string, image string) string {
+	return fmt.Sprintf(`
+resource "aws_sagemaker_model" "foo" {
+	name = "terraform-testacc-sagemaker-model-%s"
+	execution_role_arn = "${aws_iam_role.foo.arn}"
+
+	primary_container {
+		image = "%s"
+
+		environment {
```

Comment:
An equals sign will be required here in Terraform 0.12 and is backwards compatible with Terraform 0.11.  For operators, this configuration update will be detected with the `terraform 0.12upgrade` command.

```hcl
  environment = {
```